### PR TITLE
Create a new collection for hack resources

### PR DESCRIPTION
**Pull request by trevorknight**
This commit creates a new Jekyll collection for the website for
the purpose of collecting and displaying resources: software,
tutorials, or any sort of useful tool or guide.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -43,10 +43,18 @@ defaults:
...
 
 collections:
   events:
     output: true
     permalink: /:year-:month-:day.html
   posts:
     permalink: /:year-:month-:day-:title.html
+  resources:
+    output: true
+    permalink: /:title.html
```
Comment (therewasaguy) [inline _config.yml]: These links should work without the `.html` extension, right? I think we can remove the `html` extension from all of these links.

Should we reserve that first level of the permalink for the type of link, for example `resource/:title`?

**Review by therewasaguy — COMMENTED** [review]
Looks good to me, @trevorknight! Just one small comment/question

File: _config.yml (hunk 1 above)
Comment (trevorknight) [reply _config.yml]: I just added another commit that moves the main resources page to 
resources/index.html
and individual resources to 
resources/tonejs.html.

Are you asking for the resources to be
resources/tonejs/index.html?

Comment (therewasaguy) [reply _config.yml]: I meant `resources/tonejs` without the `.html` extension

Comment (trevorknight) [reply _config.yml]: Well...
Right now it is /resources/tonejs/index.html which looks like /resources/tonejs/ in the browser but for it to be /resources/tonejs will require url rewriting which IMHO is way more work than necessary. Let me know if you know a better way.

Comment (therewasaguy) [reply _config.yml]: Ah, I'm learning about Jekyll. So adding a `/` at the end of a permalink creates a folder with an index.html inside of it? That does seem like overkill.

it seems the GitHub Pages server is configured to work with or without the .html extension. For example both of these already work:

https://musichackathon.github.io/mmh_jekyll/2017-02-18
https://musichackathon.github.io/mmh_jekyll/2017-02-18.html

...so I think it's just a matter of which one we want to link to and want search engines to index. And probably better to leave out the .html extension?

I'd be happy to address permalinks in a separate PR. If you wanna squash and/or merge this it seems good to me!

**Comment by khoeger** [comment]
I like it! Can we make the tags visible beneath the tone.js on the resources page, or one day make it searchable so that we can find resources that, say, use javascript?

My main question is if it is easy to modify the layout to display resources listed by type (such as, if librosa was added, it's not really the same sort of library as tone.js, or even the pdf resources which just help link colors with sounds, and provide historical information by Candystations 2 hackathons ago. )

I think for right now, it's great to just list them, but I think in the future we may want to expand beyond a super extensive list.

Thanks for building this, Trevor!

**Comment by trevorknight** [comment]
Yiup, in the future, definitely at least grouped by tag on the Resources page and probably a clickable list of tags on the main page and on each resource page.

Rome wasn't built in a day! :)

**Review by khoeger — APPROVED** [review]
It looks good for a first stab at the resources page.

**Comment by trevorknight** [comment]
Ooof, burn. OK ok ok I'll work on improving it ASAP 😄

**Pull request merged**